src/lib.rs

```diff
@@ -210,14 +210,21 @@ impl FormatReport {
         if !new_errors.is_empty() {
             errs.has_formatting_errors = true;
         }
-        if errs.has_operational_errors && errs.has_check_errors {
+        if errs.has_operational_errors && errs.has_check_errors && errs.has_missing_errors {
```

Comment:
This would be a behavior change (e.g. license errors would no longer be reported if there was a lost comment error first) which is not necessary to address the issue at hand, and I don't want to bundle a behavior change as part of fixing the bug

```suggestion
        if errs.has_operational_errors && errs.has_check_errors {
```

Reply:
I understand this `if` as performance a optimization.  That is, if all error flags are already set then there is no need to perform the loop over the errors (now 3 flags are handled in the loop: `has_operational_errors`, `has_missing_errors`, `has_check_errors`).

If I understand the code correctly, the suggested change will cause to miss `has_missing_errors` in case both `has_operational_errors` and `has_check_errors` are already set.  It might be that with three flags now, it is better just to delete this `if` statement.

Reply:
sorry, did a quick pass through this and was not paying close enough attention here. we will indeed need this :+1: